packages/evm/src/errors.ts

```diff
@@ -38,12 +41,38 @@ export enum ERROR {
   INVALID_PROOF = 'kzg proof invalid',
 }
 
-export class EvmError {
-  error: ERROR
-  errorType: string
+export enum EvmErrorCode {
+  UNSUPPORTED_FEATURE = 'EVM_ERROR_UNSUPPORTED_FEATURE',
+  RUNTIME_ERROR = 'EVM_ERROR_RUNTIME_ERROR',
+}
+
+type EvmRuntimeErrorType = {
+  code: EvmErrorCode.RUNTIME_ERROR
+  reason: RuntimeErrorMessage | EOFError
+} & (
+  | { reason: RuntimeErrorMessage.REVERT; revertBytes: Uint8Array }
+  | { reason: Exclude<RuntimeErrorMessage, RuntimeErrorMessage.REVERT> | EOFError }
+)
```

Comment:
Is this scalable? I'm wondering how this would look if we had multiple error codes with corresponding different formats. Right now we're just having one case for "RuntimeErrorMessage.REVERT" and another excluding "RuntimeErrorMessage.REVERT", but I wonder if that might become too burdensome if we need to exclude more

Reply:
This was some experimenting with the typescript typing I did. It ensures that if `reason` is `RuntimeErrorMessage.REVERT` it has the mandatory `revertBytes`. If is NOT a `REVERT` then the reason could be any of the `RuntimeErrorMessage` or an EOFError. 

Note: EOFError has not been migrated to the EvmError, likely when I do that this typing (the second line) does not have to be there (will check)